### PR TITLE
chore: implement email edit form test

### LINKED ISSUE #1150: Create src/components/popups/profile-settings component test

### PR DESCRIPTION
**Pull request by serapieTuyishime**
# Submit a pull request

- [ ] This is not a duplicate of an existing pull request.
- [ ] No existing features have been broken without good reason.
- [ ] Your commit messages are detailed
- [ ] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

Provide details about your pull request and what it adds, fixes, or changes.

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Closes #1150 #1221

1. :question:

## Other Relevant Information

Provide any other important details below.

### CODE REVIEW

File: __tests__/components/popus/EmailForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    it("should not modify the input values of the form", async () => {
+        const emailInput = screen.getByPlaceholderText('login-page.email.placeholder');
+        const emailInputConfirm = screen.getByPlaceholderText('profile.settings.edit.email.confirm')
+
+        act(() => {
+            fireEvent.change(emailInput, { target: { value: "email1@gmail.com" } });
+            fireEvent.change(emailInputConfirm, { target: { value: "email2@gmail.com" } });
+            fireEvent.submit(screen.getByTestId("edit-email-form"));
+        })
+    });
```
[the remark this answers is not in the document]
Comment (serapieTuyishime) [reply __tests__/components/popus/EmailForm.test.tsx:64]: @coderabbitai draft the tets case to verify form submission validation logic.

File: src/components/popups/profile-settings/EmailForm.tsx [hunk 2]
```diff
@@ -108,11 +109,14 @@ export default function EditEmail({ show, onClose }: EditProfileProps): ReactEle
                   value: 2,
                   message: "This must be a valid email address",
                 },
+                validate: (val: string) => {
+                  if (watch('email') != val) return "Emails should match."
```
Comment (sezeranoJchrisostome) [inline src/components/popups/profile-settings/EmailForm.tsx:113]: Strictly comparison is more reliable.

```suggestion
                  if (watch('email') !== val) return "Emails should match."
```

File: __mocks__/colors.tsx
Comment (sezeranoJchrisostome) [inline __mocks__/colors.tsx:1]: I would suggest leaving one line at the end of file, here is the closest reference that talks about what means and why it is important.

https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

File: __tests__/components/popus/EmailForm.test.tsx
Comment (sezeranoJchrisostome) [inline __tests__/components/popus/EmailForm.test.tsx:1]: Here as well, I would suggest leaving one new line at the end of file.

**Review by sezeranoJchrisostome — COMMENTED** [review]
Looks good on my end, I left minor comment. PTAL

File: src/components/popups/profile-settings/EmailForm.tsx [hunk 3]
```diff
@@ -55,8 +56,7 @@ export default function EditEmail({ show, onClose }: EditProfileProps): ReactEle
   const onSave = async (form: FormValues) => {
     setLoading(true);
     try {
-      const { email, emailConfirm } = form;
-      if (email !== emailConfirm) return;
```
Comment (Odile078) [inline src/components/popups/profile-settings/EmailForm.tsx:59]: why was this check removed?

File: __tests__/components/popus/EmailForm.test.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        const emailInput = screen.getByPlaceholderText('login-page.email.placeholder');
+        const emailInputConfirm = screen.getByPlaceholderText('profile.settings.edit.email.confirm')
+        const closeButton = screen.getByTestId('close-button')
+
+        expect(emailInput).toBeInTheDocument()
+        expect(emailInputConfirm).toBeInTheDocument()
+        expect(closeButton).toBeInTheDocument()
+    })
+
+    it("should not modify the input values of the form", async () => {
```
Comment (Odile078) [inline __tests__/components/popus/EmailForm.test.tsx:55]: let's change the wording here, currently, it sounds like our inputs are not modifiable and that's what we are testing

File: src/components/popups/profile-settings/EmailForm.tsx (hunk 3 above)
Comment (serapieTuyishime) [reply src/components/popups/profile-settings/EmailForm.tsx:59]: the comparison is now being handled by the react-hook-form validation on line 112-114

**Review by Odile078 — APPROVED** [review]

**Review by musayann — APPROVED** [review]

**Pull request merged**